From 82e4fed602061cc7b240c3f25fbafa0c1df1c166 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Fri, 26 Feb 2021 19:09:52 +0530
Subject: [PATCH 01/23] octeontx-af: Add debug info on Mac address change

commit eb33181bb8d466b4db207d8aaf5a09322c560ffd from
git@git.assembla.com:cavium/WindRiver.linux.git

AF driver rejects change request of mac address from vf
if its set by admin(PF). Add debug info for the same.

Change-Id: I816256c56a9fffd81aad9d951af64c9fdbe4e3e8
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46831
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52273
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 814a7f336cfd..c2670378ae52 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -3383,8 +3383,11 @@ int rvu_mbox_handler_nix_set_mac_addr(struct rvu *rvu,
 	pfvf = rvu_get_pfvf(rvu, pcifunc);
 
 	/* VF can't overwrite admin(PF) changes */
-	if (from_vf && pfvf->pf_set_vf_cfg)
+	if (from_vf && pfvf->pf_set_vf_cfg) {
+		dev_warn(rvu->dev,
+			 "MAC address set by admin(PF) cannot be overwritten by VF");
 		return -EPERM;
+	}
 
 	ether_addr_copy(pfvf->mac_addr, req->mac_addr);
 
-- 
2.31.1

